### PR TITLE
clean part code under c++ grammar

### PR DESCRIPTION
**Pull request by inuoya**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by wjwithagen** [comment]
Clang seems happy about this....
However it finds  a quite a few more....

I'll submit a PR for that as well.

**Review by wjwithagen — APPROVED** [review]
Clang thinks it is oke.

File: src/rgw/rgw_aio_throttle.cc [hunk 1]
```diff
@@ -146,7 +146,7 @@ AioResultList YieldingAioThrottle::get(const RGWSI_RADOS::Obj& obj,
 
     // register the pending write and initiate the operation
     pending.push_back(*p);
-    std::move(f)(this, *static_cast<AioResult*>(p.get()));
+    f(this, *static_cast<AioResult*>(p.get()));
```
Comment (adamemerson) [inline src/rgw/rgw_aio_throttle.cc:149]: The rvalues are supposed to be this way, indicating that the functions are to be called once and should be called in move context.

Is this to fix a compiler warning or something?

**Review by adamemerson — CHANGES_REQUESTED** [review]

File: src/rgw/rgw_aio.h [hunk 2]
```diff
@@ -73,7 +73,7 @@ inline int check_for_errors(const AioResultList& results) {
 // each call returns a list of results from prior completions
 class Aio {
  public:
-  using OpFunc = fu2::unique_function<void(Aio*, AioResult&) &&>;
+  using OpFunc = fu2::unique_function<void(Aio*, AioResult&)>;
```
Comment (adamemerson) [inline src/rgw/rgw_aio.h:76]: This is part of the contract for the interface, specifically a function we will:
1. Never copy (thus it doesn't need to be copyable)
2. Only call once. That's basically the point of having the call operator rvalue qualified.

**Review by adamemerson — CHANGES_REQUESTED** [review]

**Comment by tchaikov** [comment]
@inuoya ping?

**Comment by inuoya** [comment]
@tchaikov @adamemerson 
There's no compiler warning on my Linux PC.
My roommate asked me what's going on with the usage of std::move here. It will be easier for me and my classmate to understand it after applying this PR.
I agree with the "only call once usage" here, but there's no efficiency difference between the original code and this PR. What's about using this PR to make it easier to understand the code and C++?

**Comment by tchaikov** [comment]
> It will be easier for me and my classmate to understand it after applying this PR.

hi @inuoya , i disagree. if something is just misleading or wrong what's the point of understand it? using `std::move()` follows the contract and helps to avoid the further confusion.

**Comment by inuoya** [comment]
@tchaikov 
1. "misleading" may be true. The original code requires that "f" could be only called once. If using this PR, "f" could be called more than once.
2. This PR doesn't result in "function wrong". As I've said there's no efficiency difference.

**Comment by tchaikov** [comment]
> 2. This PR doesn't result in "function wrong". As I've said there's no efficiency difference.

i think we are aiming at correctness and readability also. nothing wrong if you rewrite the whole thing in assembly, probably it can function exactly the same way, but is it "correct" thing to do? or does the rewrite make the software a better one? i doubt.

i am closing this PR, please reopen it if you disagree.

**Pull request closed without merging**